admin/proc-running-rhdh-behind-a-proxy.adoc

```diff
@@ -0,0 +1,94 @@
+[id='proc-running-rhdh-behind-a-proxy_{context}']
+= Running the {product-very-short} application behind a corporate proxy
+
+You can run the {product-very-short} application behind a corporate proxy by setting any of the following environment variables before starting the application:
+
+* `HTTP_PROXY`: Denotes the proxy to use for HTTP requests.
+* `HTTPS_PROXY`: Denotes the proxy to use for HTTPS requests.
+
+Additionally, you can set the `NO_PROXY` environment variable to exclude certain domains from proxying. The variable value is a comma-separated list of hostnames that do not require a proxy to get reached, even if one is specified.
+
+.Prerequisites
+
+* You have installed the {product}. For more information, see xref:proc-install-rhdh_rhdh-getting-started[].
+* You have created a project in the {product-short}.
+
+.Procedure
+
+. Configure proxy information in one of the following deployments based on your needs:
++
+.Helm deployment
+
+* Set the proxy information in your Helm configuration file:
++
+[source,yaml]
+----
+upstream:
+  backstage:
+    extraEnvVars:
+      - name: HTTP_PROXY
+        value: '<http_proxy_url>'
+      - name: HTTPS_PROXY
+        value: '<https_proxy_url>'
+      - name: NO_PROXY
+        value: '<no_proxy_settings>'
+----
++
+Where,
+
+`<http_proxy_url>`:: Denotes a variable that you must replace with the HTTP proxy URL.
+`<https_proxy_url>`:: Denotes a variable that you must replace with the HTTPS proxy URL.
+`<no_proxy_settings>`:: Denotes a variable that you must replace with comma-separated URLs, which you want to exclude from proxying, for example, `foo.com,baz.com`.
++
+.Example: Setting proxy variables using Helm Chart
+
+[source,yaml]
+----
+upstream:
+  backstage:
+    extraEnvVars:
+      - name: HTTP_PROXY
+        value: 'http://10.10.10.105:3128'
+      - name: HTTPS_PROXY
+        value: 'http://10.10.10.106:3128'
+      - name: NO_PROXY
+        value: 'localhost,example.org'
+----
+
++
+.Operator deployment 
+
+** Perform one of the following steps:
+* Set the proxy information in a ConfigMap file that contains default configuration:
++
+.. Search for a ConfigMap file named `backstage-default-config` in the namespace where the Operator is deployed and open it. 
+.. Find the `deployment.yaml` key.
+.. Add the `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY` environment variables to the containers listed in the `Deployment` spec as shown in the following example:
++ 
+.Example: Setting proxy variables in a ConfigMap file
+[source,yaml]
+----
+
+
+----
```

Comment:
@rm3l, Could you please share a yaml example for setting proxy variables in a configmap file?

Reply:
Well, the default config of the operator is quite a large ConfigMap, which I'm attaching [here](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/files/15333866/rhdh-operator--backstage-default-config-1.2.yaml.txt) if you want, but it can be fetched by running the following command:
```shell
$ oc get configmap backstage-default-config -n rhdh-operator -o yaml > rhdh-operator-default-config.yaml
```

For your convenience, here is the diff with the proxy settings added to the init and main container of the RHDH application pod listed in the operator default config ConfigMap:
```diff
diff --git a/rhdh-operator--backstage-default-config-1.2.yaml b/rhdh-operator--backstage-default-config-1.2.yaml
index 99eea5d..30a7367 100644
--- a/rhdh-operator--backstage-default-config-1.2.yaml
+++ b/rhdh-operator--backstage-default-config-1.2.yaml
@@ -211,6 +211,12 @@ data:
               env:
                 - name: NPM_CONFIG_USERCONFIG
                   value: /opt/app-root/src/.npmrc.dynamic-plugins
+                - name: HTTP_PROXY
+                  value: '<http_proxy_url>'
+                - name: HTTPS_PROXY
+                  value: '<https_proxy_url>'
+                - name: NO_PROXY
+                  value: '<no_proxy_settings>'
               volumeMounts:
                 - mountPath: /dynamic-plugins-root
                   name: dynamic-plugins-root
@@ -264,6 +270,12 @@ data:
               env:
                 - name: APP_CONFIG_backend_listen_port
                   value: "7007"
+                - name: HTTP_PROXY
+                  value: '<http_proxy_url>'
+                - name: HTTPS_PROXY
+                  value: '<https_proxy_url>'
+                - name: NO_PROXY
+                  value: '<no_proxy_settings>'
               volumeMounts:
                 - mountPath: /opt/app-root/src/dynamic-plugins-root
                   name: dynamic-plugins-root
```